src/Parser.js

```diff
@@ -758,6 +764,12 @@ export default class Parser {
             return this.parseUrlGroup(optional);
         }
```

Comment:
Maybe `defineGroupParser` can be used for `color`, `size`, `url`, and even `original`.